partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerGraphMatchTests.kt

```diff
@@ -14,7 +14,292 @@ class EvaluatingCompilerGraphMatchTests : EvaluatorTestBase() {
+                |                   {id: d2, payload: 11.11, ends: (n1 -> n1)} ] }""".trimMargin(),
+            // n1  n2  -- two disconnected nodes
+            "N2E0" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], edges: [] }""",
+            // n1 -[d1]-> n2
+            "N2D1" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], 
+                |          edges: [ {id: d1, payload: 1.2, ends: (n1 -> n2) } ] }""".trimMargin(),
+            // n1 -[u1]- n2
+            "N2U1" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], 
+                |          edges: [ {id: u1, payload: 1.2, ends: (n1 -- n2) } ] }""".trimMargin(),
+            // n1 -[d1]-> n2   --- two parallel edges
+            //    -[d2]->
+            "N2D2" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], 
+                |          edges: [ {id: d1, payload: 1.2,   ends: (n1 -> n2) }, 
+                |                   {id: d2, payload: 11.22, ends: (n1 -> n2) } ] }""".trimMargin(),
+            // n1 -[d1]-> n2   --- two cycling edges
+            //    <-[d2]-
+            "N2D2c" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], 
+                |           edges: [ {id: d1, payload: 1.2, ends: (n1 -> n2) },
+                |                    {id: d2, payload: 2.1, ends: (n2 -> n1) } ] }""".trimMargin(),
+            // n1 -[u1]- n2   --- two parallel undirected edges
+            //    -[u2]-
+            "N2U2" to """{ nodes: [ {id: n1, payload: 1}, {id: n2, payload: 2} ], 
+                |          edges: [ {id: u1, payload: 1.2, ends: (n1 -- n2) },
+                |                   {id: u2, payload: 2.1, ends: (n2 -- n1) } ] }""".trimMargin(),
+        )
+    )
+
+    @ParameterizedTest
+    @ArgumentsSource(SmallGraphs::class)
+    fun testSmallGraphs(qr: Pair<String, String>) {
+        testGraphQueries(sessionSmallGraphs, qr)
+    }
+
+    class SmallGraphs : ArgumentsProviderBase() {
+        override fun getParameters(): List<Pair<String, String>> = listOf(
+            "(N0E0 MATCH (x))" to "<< >>",
+            "(N0E0 MATCH -[y]-> )" to "<< >>",
+            "(N0E0 MATCH (x)-[y]->(z) )" to "<< >>",
+
+            "(N1E0 MATCH (x))" to "<< {'x':1} >>",
+            "(N1E0 MATCH -[y]-> )" to "<< >>",
+            "(N1E0 MATCH (x)-[y]->(z) )" to "<< >>",
+            "(N1E0 MATCH (x)-[y]->(x) )" to "<< >>",
+
+            "(N1D1 MATCH (x))" to "<< {'x':1} >>",
+            "(N1D1 MATCH -[y]-> )" to "<< {'y':1.1} >>",
+            "(N1D1 MATCH (x)-[y]->(z) )" to "<< {'x':1, 'y':1.1, 'z':1} >>",
+            "(N1D1 MATCH (x)-[y]->(x) )" to "<< {'x':1, 'y':1.1} >>",
+            "(N1D1 MATCH (x1)-[y1]->(x2)-[y2]->(x3) )" to "<< {'x1':1, 'y1':1.1, 'x2':1, 'y2':1.1, 'x3':1} >>",
+
+            "(N1U1 MATCH (x))" to "<< {'x':1} >>",
+//  Results are not yet correct: need the deduplication of Section 6.5
+//            "(N1U1 MATCH ~[y]~ )" to "<< {'y':1.1} >>",
```

Comment:
Noticed there are some conformance tests failing when using the `partiql-tests` commit w/ the new graph tests (see [conformance report comment](https://github.com/partiql/partiql-lang-kotlin/pull/1104#issuecomment-1563486956)).

Just checking do those failing tests correspond to the commented out tests in this file? The number of failing matches up (12 test failing = 6 commented out tests * 2 evaluation modes tested)

Reply:
Yes, those should be due to the commented-out unit tests -- I decided to include them as conformance tests, as our setup is "resilient" to those failing.  

(But if I am lucky, these failures will not last for too long.)